staging/src/k8s.io/component-base/logs/options_test.go

```diff
@@ -99,3 +104,42 @@ func TestOptions(t *testing.T) {
 		})
 	}
 }
+
+func TestContextualLogging(t *testing.T) {
+	t.Run("enabled", func(t *testing.T) {
+		testContextualLogging(t, true)
+	})
+
+	t.Run("disabled", func(t *testing.T) {
+		testContextualLogging(t, false)
+	})
+}
+
+func testContextualLogging(t *testing.T, enabled bool) {
+	var err error
+
+	o := NewOptions()
+	featureGate := featuregate.NewFeatureGate()
+	AddFeatureGates(featureGate)
+	err = featureGate.Set(fmt.Sprintf("ContextualLogging=%v", enabled))
```

Comment:
prefer SetFromMap so we can use the ContextualLogging constant and get compile-time checks?

Reply:
Done.